executor/aggregate.go

```diff
@@ -660,20 +660,54 @@ func (e *HashAggExec) waitFinalWorkerAndCloseFinalOutput(waitGroup *sync.WaitGro
 
 func (e *HashAggExec) prepare4ParallelExec(ctx context.Context) {
 	go e.fetchChildData(ctx)
-
+	partialTime := make(chan time.Duration, e.ctx.GetSessionVars().HashAggPartialConcurrency())
 	partialWorkerWaitGroup := &sync.WaitGroup{}
 	partialWorkerWaitGroup.Add(len(e.partialWorkers))
+	partialStart := time.Now()
 	for i := range e.partialWorkers {
-		go e.partialWorkers[i].run(e.ctx, partialWorkerWaitGroup, len(e.finalWorkers))
-	}
-	go e.waitPartialWorkerAndCloseOutputChs(partialWorkerWaitGroup)
-
+		go func(index int) {
+			workerStart := time.Now()
+			e.partialWorkers[index].run(e.ctx, len(e.finalWorkers))
+			if e.stats != nil {
+				partialTime <- time.Since(workerStart)
+			}
```

Comment:
How about removing this channel and setting `e.stats.PartialTime[index] = xxxx`?